lib/rex/proto/http/client.rb

```diff
@@ -144,11 +144,40 @@ def request_raw(opts = {})
   def request_cgi(opts = {})
     opts = self.config.merge(opts)
 
+    opts['data'] ||= ''
     opts['cgi'] = true
     opts['port'] = self.port
     opts['ssl'] = self.ssl
+    opts['ctype'] ||= 'multipart/form-data; boundary=0' if opts['files']
     opts['ctype'] ||= 'application/x-www-form-urlencoded' if opts['method'] == 'POST'
 
+    if opts['files'] && opts['files'].is_a?(::Array) && opts['method'] == 'POST'
+
+      file_data = Rex::MIME::Message.new
+
+      opts['files'].each do |file_hash|
+        # Set default values
+        mime_type = file_hash[:mime_type] || 'application/octet-stream'
+        encoding = file_hash[:encoding] || nil
+        content_disposition = file_hash[:content_disposition] || nil
```

Comment:
I think we can generate this entirely now, so it's very rare a client will ever have to specify it - but leaving it in as an override works for me - your call :+1: